### PR TITLE
EX-3448 Update Adapter to support video

### PR DESCRIPTION
**Pull request by maximpakhomov**

### CODE REVIEW

**Review by mikhalovich — COMMENTED** [review]
@jrosendahl Should we add validation for required fields [here](https://github.com/sovrn/Prebid.js/pull/36/files#diff-47700f8dab978f7908ef5969dd8043b8fde6ee51690e7f80a9d19a387f34ec3bL17)?

File: modules/sovrnBidAdapter.js [hunk 1]
```diff
@@ -50,27 +53,48 @@ export const spec = {
...
+          imp.banner = {
             format: processedSizes,
             w: 1,
             h: 1,
-          },
-          tagid: String(utils.getBidIdParameter('tagid', bid.params)),
-          bidfloor: floorInfo.floor
+          };
+        }
+        if (bid.mediaTypes && bid.mediaTypes.video) {
```
Comment (jrosendahl) [inline modules/sovrnBidAdapter.js]: we need to not support bid.mediaTypes.video.context = 'adpod'

File: modules/sovrnBidAdapter.js [hunk 2]
```diff
@@ -158,7 +182,7 @@ export const spec = {
             dealId: sovrnBid.dealid || null,
             currency: 'USD',
             netRevenue: true,
```
Comment (jrosendahl) [inline modules/sovrnBidAdapter.js:196]: The returned VAST URL or raw VAST XML should be added into bid.vastUrl or bid.vastXml, respectively.

File: modules/sovrnBidAdapter.js [hunk 3]
```diff
@@ -50,27 +53,48 @@ export const spec = {
...
             format: processedSizes,
             w: 1,
             h: 1,
-          },
-          tagid: String(utils.getBidIdParameter('tagid', bid.params)),
-          bidfloor: floorInfo.floor
+          };
+        }
+        if (bid.mediaTypes && bid.mediaTypes.video) {
+          let bidSizes = bid.mediaTypes.video.playerSize
```
Comment (jrosendahl) [inline modules/sovrnBidAdapter.js]: This could be done better by getting video first (line 87).

File: modules/sovrnBidAdapter.js [hunk 4]
```diff
@@ -50,27 +53,48 @@ export const spec = {
...
+          bidSizes = (utils.isArray(bidSizes) && utils.isArray(bidSizes[0])) ? bidSizes : [bidSizes]
+          bidSizes = bidSizes.filter(size => utils.isArray(size))
+
+          const video = utils.deepAccess(bid, 'mediaTypes.video') || {}
+
+          imp.video = {
```
Comment (jrosendahl) [inline modules/sovrnBidAdapter.js]: We should support all of the video parameters 
I really like how this adapter did it https://github.com/prebid/Prebid.js/blob/832c13ee4126ac20a81e00fc35c8d71fa0afff25/modules/deepintentBidAdapter.js
specifically 
https://github.com/prebid/Prebid.js/blob/832c13ee4126ac20a81e00fc35c8d71fa0afff25/modules/deepintentBidAdapter.js#L179

File: test/spec/modules/sovrnBidAdapter_spec.js [hunk 5]
```diff
@@ -95,7 +161,10 @@ describe('sovrnBidAdapter', function() {
         'params': {
           'iv': 'vet'
         },
-        'sizes': [300, 250]
+        'sizes': [300, 250],
+        'mediaTypes': {
+          banner: {}
+        },
       }
       const request = spec.buildRequests([singleSizeBidRequest], baseBidderRequest)
```
Comment (jrosendahl) [inline test/spec/modules/sovrnBidAdapter_spec.js:148]: we probably need an interpretresponse test as well

File: modules/sovrnBidAdapter.js [hunk 6]
```diff
@@ -158,7 +182,7 @@ export const spec = {
             dealId: sovrnBid.dealid || null,
             currency: 'USD',
             netRevenue: true,
-            mediaType: BANNER,
```
Comment (jrosendahl) [inline modules/sovrnBidAdapter.js:161]: Also for video we are going to not send the nurl, we will have it embedded in the creative.

**Review by jrosendahl — CHANGES_REQUESTED** [review]

File: modules/sovrnBidAdapter.js (hunk 2 above)
Comment (jrosendahl) [reply modules/sovrnBidAdapter.js:196]: We only ever will only have VAST XML

File: modules/sovrnBidAdapter.js [hunk 7]
```diff
@@ -158,11 +194,21 @@ export const spec = {
...
             ttl: sovrnBid.ext ? (sovrnBid.ext.ttl || 90) : 90,
             meta: { advertiserDomains: sovrnBid && sovrnBid.adomain ? sovrnBid.adomain : [] }
-          });
+          }
+
+          const videoRegex = new RegExp(/VAST\s+version/)
+
+          if (videoRegex.test(sovrnBid.adm)) {
+            bid.mediaType = VIDEO
+            bid.ad = decodeURIComponent(sovrnBid.adm)
```
Comment (jrosendahl) [inline modules/sovrnBidAdapter.js]: I am unsure if we want to set ad for video units, we probably want to decodeURI before the regex

Comment (jrosendahl) [reply modules/sovrnBidAdapter.js]: And we probably want to have uri decoded data in the VAST XML field

Comment (jrosendahl) [reply modules/sovrnBidAdapter.js]: Do we want to rely on a regex or should we reply with the type in our response @cpabst

File: modules/sovrnBidAdapter.js [hunk 8]
```diff
@@ -1,12 +1,38 @@
 import * as utils from '../src/utils.js'
 import { registerBidder } from '../src/adapters/bidderFactory.js'
-import { BANNER } from '../src/mediaTypes.js'
+import { ADPOD, BANNER, VIDEO } from '../src/mediaTypes.js'
 import { createEidsArray } from './userId/eids.js';
 import {config} from '../src/config.js';
 
+export const ORTB_VIDEO_PARAMS = {
```
Comment (jrosendahl) [inline modules/sovrnBidAdapter.js]: this probably does not need to be exported

**Review by jrosendahl — CHANGES_REQUESTED** [review]
Looking better, some comments

**Review by jrosendahl — APPROVED** [review]
Looks great but Do not submit to prebid until integration tests have been ran.

**Review by cpabst — APPROVED** [review]
Integration tests have been run.

**Comment by cpabst** [comment]
Close after prebid merge

**Pull request closed without merging**